src/xcode/ENA/ENA/Source/Scenes/ExposureDetection/Survey/SurveyConsentViewController.swift

```diff
@@ -40,14 +40,19 @@ final class SurveyConsentViewController: DynamicTableViewController, ENANavigati
 	// MARK: - Protocol ENANavigationControllerWithFooterChild
 
 	func navigationController(_ navigationController: ENANavigationControllerWithFooter, didTapPrimaryButton button: UIButton) {
-		guard let url = URL(string: "https://www.test.de") else {
-			return
+		navigationFooterItem?.isPrimaryButtonLoading = true
+
+		viewModel.getURL { [weak self] result in
+			switch result {
+			case .success(let url):
+				self?.completion(url)
+			case .failure(let error):
+				self?.showErrorAlert(with: error)
```

Comment:
Would be nice to log the error

Reply:
Our does the Service already do this?

Reply:
The logs should be at the service layers.

Reply:
I'll add them.